dessia_common/models/workflows/forms_workflow.py

```diff
@@ -3,22 +3,23 @@
 from dessia_common.workflow.blocks import InstantiateModel, ModelMethod, ModelAttribute, ForEach,\
     MultiPlot, Unpacker, WorkflowBlock
 from dessia_common.forms import Generator, Optimizer
-from dessia_common.typings import MethodType
+from dessia_common.typings import MethodType, AttributeType
 
 instanciate_generator = InstantiateModel(model_class=Generator, name='Instantiate Generator')
 
 generate_method = MethodType(class_=Generator, name='generate')
 generator_generate = ModelMethod(method_type=generate_method, name='Generator Generate')
-attribute_selection = ModelAttribute(attribute_name='models', name='Attribute Selection')
+generate_attribute = AttributeType(class_=Generator, name='models')
+attribute_selection = ModelAttribute(attribute_type=generate_attribute, name='Attribute Selection')
 
 # Sub-Workflow of model optimization
 instanciate_optimizer = InstantiateModel(model_class=Optimizer, name='Instantiate Optimizer')
 
 
 generate_method = MethodType(class_=Optimizer, name='optimize')
 optimization = ModelMethod(method_type=generate_method, name='Optimization')
-
-model_fetcher = ModelAttribute(attribute_name='model_to_optimize', name='Model Fetcher')
+generate_attribute = AttributeType(class_=Optimizer, name='model_to_optimize')
```

Comment:
Variable name seems a bit off the context. Is this a copy/paste from several lines before ?

Reply:
Yes, I've renamed it model_to_optimize_attribute